orchestrator/graphql/resolvers/process.py

```diff
@@ -0,0 +1,224 @@
+                        value_as_uuid = UUID(value)
+                    except (ValueError, AttributeError):
+                        msg = "Not a valid customer_id, must be a UUID: '{value}'"
+                        logger.exception(msg)
+                        raise_status(HTTPStatus.BAD_REQUEST, msg)
+                    process_subscriptions = (
+                        db.session.query(ProcessSubscriptionTable)
+                        .join(SubscriptionTable)
+                        .filter(SubscriptionTable.customer_id == value_as_uuid)
+                        .subquery()
+                    )
+                    query = query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+                elif field == "product":
+                    process_subscriptions = (
+                        db.session.query(ProcessSubscriptionTable)
+                        .join(SubscriptionTable, ProductTable)
+                        .filter(ProductTable.name.ilike("%" + value + "%"))
+                        .subquery()
+                    )
+                    query = query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+                elif field == "tag":
+                    tags = value.split("-")
+                    process_subscriptions = (
+                        db.session.query(ProcessSubscriptionTable)
+                        .join(SubscriptionTable, ProductTable)
+                        .filter(ProductTable.tag.in_(tags))
+                        .subquery()
+                    )
+                    query = query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+                elif field == "subscriptions":
+                    process_subscriptions = (
+                        db.session.query(ProcessSubscriptionTable)
+                        .join(SubscriptionTable)
+                        .filter(SubscriptionTable.description.ilike("%" + value + "%"))
+                        .subquery()
+                    )
+                    query = query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+                elif field == "pid":
+                    query = query.filter(cast(ProcessTable.pid, String).ilike("%" + value + "%"))
+                elif field == "target":
+                    targets = value.split("-")
+                    process_subscriptions = (
+                        db.session.query(ProcessSubscriptionTable)
+                        .filter(ProcessSubscriptionTable.workflow_target.in_(targets))
+                        .subquery()
+                    )
+                    query = query.filter(ProcessTable.pid == process_subscriptions.c.pid)
+                else:
+                    raise_status(HTTPStatus.BAD_REQUEST, f"Invalid filter '{field}'")
+    return query
+
+
+def sort_processes(query: SearchQuery, sort_by: list[ProcessSort] | None = None) -> SearchQuery:
+    if sort_by is not None:
+        for item in sort_by:
+            if item.field in VALID_SORT_KEYS:
```

Comment:
I would suggest to split this in a part that does the validation and another part that builds the query instead of 2 things in a single for loop.

Reply:
added generic sorting functionality for sqlalchemy.